app/Models/User.php

```diff
@@ -472,6 +472,11 @@ public static function lookupWithHistory($usernameOrId, $type = null, $findAll =
             return $user;
         }
 
+        // don't perform username change history lookup if we're searching by ID
+        if (ctype_digit((string) $usernameOrId)) {
+            return null;
+        }
```

Comment:
Usernames and name changes can be all digits; when it's an id-only lookup then then there's no point in doing the username history lookup since it's not looking for a name.